include/boost/decimal/detail/cmath/log.hpp

```diff
@@ -164,29 +93,6 @@ constexpr auto log_impl(T x) noexcept
     return result;
 }
 
-} //namespace detail
-
-BOOST_DECIMAL_EXPORT template <typename T>
-constexpr auto log(T x) noexcept
-    BOOST_DECIMAL_REQUIRES(detail::is_decimal_floating_point_v, T)
-{
-    #if BOOST_DECIMAL_DEC_EVAL_METHOD == 0
-
-    using evaluation_type = T;
-
-    #elif BOOST_DECIMAL_DEC_EVAL_METHOD == 1
-
-    using evaluation_type = detail::promote_args_t<T, decimal64>;
-
-    #else // BOOST_DECIMAL_DEC_EVAL_METHOD == 2
-
-    using evaluation_type = detail::promote_args_t<T, decimal128>;
-
-    #endif
-
-    return static_cast<T>(detail::log_impl(static_cast<evaluation_type>(x)));
-}
```

Comment:
We need to keep this part on each of the special functions to properly support the float evaluation methods. This promotes the type of x depending on the macro set and then does the calculation much like how boost math does.